From 84c332b36ca16b6c27d049a1b1be0451070e0543 Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Mon, 12 Feb 2024 17:43:26 +0100
Subject: [PATCH 0787/1695] plat-stm32mp2: stm32mp2_pmic: add regulator
 configuration

Configure the default ramp delay for the STPMIC25 regulators, the
DT configuration should be use only to change this values, for example
fort boards or external device constraints.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 1c096d72dd7c4b324caa096e30d28fe52bb1dcae plat-stm32mp2: stm32mp2_pmic: add regulator configuration]
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Change-Id: I31131726c07eb20d441b3282af54a6baf87f0922
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/359384
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Pascal PAILLET <p.paillet@st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 core/arch/arm/plat-stm32mp2/drivers/stm32mp2_pmic.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/core/arch/arm/plat-stm32mp2/drivers/stm32mp2_pmic.c b/core/arch/arm/plat-stm32mp2/drivers/stm32mp2_pmic.c
index a61d7ffef..9dd4cdde2 100644
--- a/core/arch/arm/plat-stm32mp2/drivers/stm32mp2_pmic.c
+++ b/core/arch/arm/plat-stm32mp2/drivers/stm32mp2_pmic.c
@@ -438,6 +438,15 @@ static TEE_Result pmic_regu_pm(enum pm_op op, uint32_t pm_hint,
 	return apply_pm_state(regulator, mode);
 }
 
+static TEE_Result pmic_init(struct regulator *regulator)
+{
+	/* default configuration for STPMIC regulator */
+	regulator->ramp_delay_uv_per_us = U(2200);
+	regulator->enable_ramp_delay_us = U(1000);
+
+	return TEE_SUCCESS;
+}
+
 static TEE_Result pmic_supplied_init(struct regulator *regulator,
 				     const void *fdt, int node)
 {
@@ -497,6 +506,7 @@ static const struct regulator_ops pmic_regu_ops = {
 	.set_voltage = pmic_set_voltage,
 	.get_voltage = pmic_get_voltage,
 	.supported_voltages = pmic_list_voltages,
+	.init = pmic_init,
 	.supplied_init = pmic_supplied_init,
 };
 
-- 
2.47.2

